File: Programming Basics with Python/06 Nested loops/More exercises/04_Car_Number.py
interval_start = int(input())
interval_end = int(input())
for first_digit in range(interval_start, interval_end + 1):
    for second_digit in range(interval_start, interval_end + 1):
        for third_digit in range(interval_start, interval_end + 1):
            if (second_digit + third_digit) % 2 != 0:
                continue
            for fourth_digit in range(interval_start, interval_end + 1):
                if first_digit > fourth_digit:
                    if first_digit % 2 == 0:
                        if fourth_digit % 2 == 0:
                            continue
                    if first_digit % 2 == 1:
                        if fourth_digit % 2 == 1:
                            continue
                    print(f"{first_digit}{second_digit}{third_digit}{fourth_digit}", end=" ")

File: JS Basics/06 Nested Loops - More Exercises/04. Car Number.js
function carNum(input) {
    let startInterval = Number(input[0]);
    let endInterval = Number(input[1]);
    let print = "";
    for (let firstDigit = startInterval; firstDigit <= endInterval; firstDigit++) {
        for (let secondDigit = startInterval; secondDigit <= endInterval; secondDigit++) {
            for (let thirdtDigit = startInterval; thirdtDigit <= endInterval; thirdtDigit++) {
                for (let fourthDigit = startInterval; fourthDigit <= endInterval; fourthDigit++) {
                    if (((firstDigit % 2 == 0 && fourthDigit % 2 != 0) || (firstDigit % 2 != 0 && fourthDigit % 2 == 0)) && (firstDigit > fourthDigit) && ((secondDigit + thirdtDigit) % 2 == 0)) {
                        print += (`${firstDigit}${secondDigit}${thirdtDigit}${fourthDigit} `);
                    }
                }
            }
        }
    }
    console.log(print);
}
carNum([5, 8])
